drivers/include/net/netdev.h

```diff
@@ -290,6 +294,11 @@ struct netdev {
 #ifdef MODULE_L2FILTER
     l2filter_t filter[L2FILTER_LISTSIZE];   /**< link layer address filters */
 #endif
+#ifdef MODULE_NETSTATS_NEIGHBOR
+    cib_t stats_idx;                                        /**< CIB for the tx correlation */
+    netstats_nb_t *stats_queue[NETSTATS_NB_QUEUE_SIZE];     /**< send/callback mac association array */
+    netstats_nb_t pstats[NETSTATS_NB_SIZE];                 /**< Per neighbor statistics array */
+#endif
```

Comment:
In OP you said you want to add this to `gnrc_netif`, but here it is in `netdev`. Might I propose to maybe move this to `netif` directly?

Reply:
Sorry, mixed them up.
IIRC you suggested this should be independent of GNRC, so attaching this to `netdev` sounds like a good place.
We could save some memory by extending NIB, but then other protocol stacks could not use it.

Reply:
Yes, but `netdev` is on a potential way out (see @jia200x's rework efforts), so I would prefer to see this network stack independent, but future-proof in `netif_t` (not to be confused with `gnrc_netif_t` nor lwIP's `struct netif` ;-))